packages/plugin-core/src/test/suite-integ/ShowPreview.test.ts

```diff
@@ -19,7 +19,71 @@ suite("ShowPreview utility methods", () => {
     beforeHook: () => {},
   });
 
-  describe(`handleLink`, () => {
+  describe("vaultlessAssetPath", () => {
+    describe("GIVEN valid href", () => {
+      beforeEach(async () => {
+        const factory = TestNoteFactory.defaultUnitTestFactory();
+
+        await factory.createForFName("foo");
+      });
+
+      afterEach(() => {
+        sinon.restore();
+      });
+      it("THEN correct asset path is corectly decoded", (done) => {
+        runSingleVaultTest({
+          ctx,
+          onInit: async ({ wsRoot, vault }) => {
+            const handler = new PreviewLinkHandler(
+              new MockDendronExtension({
+                engine: ExtensionProvider.getEngine(),
+                wsRoot,
+                vaults: [vault],
+              })
+            );
+
+            // without urlencoded parts
+            const withoutURLEncode = handler.vaultlessAssetPath({
+              data: {
+                href: "vscode-webview://e380a62c-2dea-46a8-ae1e-a34868c9719e/assets/dummy-pdf.pdf",
+                id: "foo",
+              },
+              wsRoot,
+            });
+            expect(withoutURLEncode !== undefined).toBeTruthy();
+            expect(
+              await AssertUtils.assertInString({
+                body: withoutURLEncode as string,
+                match: ["assets/dummy-pdf.pdf"],
+              })
+            );
+
+            // with urlencoded parts (space as %20)
+            const withURLEncode = handler.vaultlessAssetPath({
+              data: {
+                href: "vscode-webview://e380a62c-2dea-46a8-ae1e-a34868c9719e/assets/file%20with%20space.pdf",
+                id: "foo",
+              },
+              wsRoot,
+            });
+            expect(withURLEncode !== undefined).toBeTruthy();
+            expect(
+              await AssertUtils.assertInString({
+                body: withURLEncode as string,
+                match: ["assets/file with space.pdf"],
+              })
+            );
+
+            done();
+          },
+        });
+      });
+    });
+  });
+
+  // this test block passes because it doesn't wait for the actual test to run.
+  // the actual test does not pass.
+  describe.skip(`handleLink`, () => {
```

Comment:
I've skipped this test because this test was pre-emptively passing before the actual code was executed.
Even with proper test resolution, this test code will fail.

Reply:
I'll try to block some time to fix this, but I don't think this would be within the scope of this PR.